src/parser/expression.js

```diff
@@ -896,6 +896,14 @@ pp.parseObj = function (isPattern, refShorthandDefaultPos) {
   node.properties = [];
   this.next();
 
+  // `for` keyword begins an object comprehension.
+  if (this.hasPlugin("lightscript") && this.match(tt._for)) {
+    // ...however, `{ for: x }` is a legal JS object.
```

Comment:
ah, that does complicate things... 

At least we dont have to deal with `myAnnoyingDestructuredObj = { for }`.

I'd almost rather just add a restriction to the language saying that if you want an object where the first key is `for`, you need to wrap it in quotes: `{ "for": "reasons" }`.

It'd be less confusing as a reader in that case anyway. Thoughts?